src/status_im/ui/components/list_item/views.cljs

```diff
@@ -1,82 +1,284 @@
+      theme subtitle content accessories])
+
+   (when (and subtitle (= :default type))
+     [subtitle-row subtitle-row-elements icon?])
+
+   (when content
+     [react/view {:margin-left (if icon? 2 0)}
+      (if (vector? content)
+        content
+        [content])])])
+
+(defn- accessories-column [accessories]
+  (let [last-accessory              (peek accessories)
+        last-accessory-is-component (and (not (string? last-accessory))
+                                         (not= :chevron last-accessory))
+        second-last-accessory       (peek (pop accessories))]
+    (vec
+     (concat
+      [react/view {:align-self :stretch
+                   :flex-direction :row
+                   :align-items :center
+                   :justify-content :flex-start}]
       (for [accessory accessories]
         (when-not (nil? accessory)
           (with-meta
             (cond
               (= :chevron accessory)
-              [react/view {:margin-right 8}
-               [icons/icon :main-icons/next {:color colors/gray-transparent-40}]]
+              [react/view
+               [icons/icon
+                :main-icons/next
+                {:container-style {:width           9
+                                   :height          16
+                                   :justify-content :center}
+                 :width           10
+                 :resize-mode     :center
+                 :height          16
+                 :color           colors/gray-transparent-40}]]
+
               (= :check accessory)
-              [react/view {:margin-right 16}
+              [react/view {:background-color colors/gray-transparent-40}
                [icons/icon :main-icons/check {:color colors/gray}]]
+
               (= :more accessory)
-              [react/view {:margin-right 16}
+              [react/view
                [icons/icon :main-icons/more]]
+
               :else
-              [react/view {:margin-right 16 :flex-shrink 1}
+              [react/view {:margin-right
+                           (if (= accessory last-accessory)
+                             0
+                             (if (and last-accessory-is-component
+                                      (string? second-last-accessory))
+                               1
+                               16))}
                (cond
                  (string? accessory)
-                 [react/text {:style styles/accessory-text :number-of-lines 1}
+                 [react/text {:style           styles/accessory-text
+                              :number-of-lines 1}
                   accessory]
+
                  (vector? accessory)
                  accessory
+
                  :else
                  [accessory])])
-            {:key accessory})))
-      (when error
-        [tooltip/tooltip error styles/error])]]))
+            {:key accessory})))))))
+
+(defn list-item
+  [{:keys [type theme container-margin-top container-margin-bottom image image-path
```

Comment:
maybe we could move styles in a separate map? `container-margin-top` , `title-color-override` etc

Reply:
yes I realized it and already done it locally

I'm making all these little cleanups everywhere for sure :)

thanks!

Reply:
Actually, I'm not so sure about styles in a separate map.

I believe the purpose and intent of the design spec, for a versatile component such as list-item, is to constrain design as much as possible, all the while allowing constrained flexibility for the purpose of consistency.

If we allow unconstrained overrides on style via separate map of styles we will likely break constraint and consistency.

I'll keep this as a TODO - to reconsider at last phase of getting this done

at the moment quite deep in the minutia that is called for in this PR 😅 and almost there with another push

Reply:
I have explained detail/purpose/context of each key of the prop in list-item component definition. Hope that it will add to the explanation that it is better to keep the prop shallow as well.